### PR TITLE
Azure deployment refactor

### PR DESCRIPTION
**Pull request by tjholm**
Refactors most resources to use azure-native (just one more to go before we can remove the plugin).

Fixes issue where subscription ID was malformed for container app deployments causing azure api client to fail.

### CODE REVIEW

File: pkg/codeconfig/codeconfig.go [hunk 1]
```diff
@@ -311,25 +310,19 @@ func (c *codeConfig) collectOne(handler string) error {
...
+		logreader, err := ce.ContainerLogs(cID, types.ContainerLogsOptions{
+			ShowStdout: true,
+			ShowStderr: true,
+			Follow:     true,
+		})
+		if err != nil {
+			return err
+		}
+		go func() {
+			_, _ = stdcopy.StdCopy(log.Writer(), log.Writer(), logreader)
```
Comment (asalkeld) [inline pkg/codeconfig/codeconfig.go]: this looks like a rebase error

File: pkg/provider/pulumi/azure/azure.go [hunk 2]
```diff
@@ -27,12 +27,14 @@ import (
 	"github.com/AlecAivazis/survey/v2"
 	"github.com/golangci/golangci-lint/pkg/sliceutil"
 	"github.com/pkg/errors"
-	"github.com/pulumi/pulumi-azure/sdk/v4/go/azure/core"
-	"github.com/pulumi/pulumi-azure/sdk/v4/go/azure/eventgrid"
-	"github.com/pulumi/pulumi-azure/sdk/v4/go/azure/keyvault"
 	"github.com/pulumi/pulumi/sdk/v3/go/auto"
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
```
Comment (asalkeld) [inline pkg/provider/pulumi/azure/azure.go]: remove this space above

File: pkg/provider/pulumi/azure/azure.go [hunk 3]
```diff
@@ -179,37 +181,49 @@ func (a *azureProvider) Deploy(ctx *pulumi.Context) error {
...
-	rg, err := core.NewResourceGroup(ctx, resourceName(ctx, "", ResourceGroupRT), &core.ResourceGroupArgs{
+	rg, err := resources.NewResourceGroup(ctx, resourceName(ctx, "", ResourceGroupRT), &resources.ResourceGroupArgs{
 		Location: pulumi.String(a.t.Region),
 		Tags:     common.Tags(ctx, ctx.Stack()),
 	})
+
+	// rg, err := core.NewResourceGroup(ctx, resourceName(ctx, "", ResourceGroupRT), &core.ResourceGroupArgs{
+	// 	Location: pulumi.String(a.t.Region),
+	// 	Tags:     common.Tags(ctx, ctx.Stack()),
+	// })
```
Comment (asalkeld) [inline pkg/provider/pulumi/azure/azure.go]: delete this above ^

File: pkg/provider/pulumi/azure/containerapp.go [hunk 4]
```diff
@@ -17,12 +17,15 @@
...
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/authorization"
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/containerregistry"
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/eventgrid"
+	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/operationalinsights"
 	web "github.com/pulumi/pulumi-azure-native/sdk/go/azure/web/v20210301"
-	"github.com/pulumi/pulumi-azure/sdk/v4/go/azure/authorization"
-	"github.com/pulumi/pulumi-azure/sdk/v4/go/azure/containerservice"
-	"github.com/pulumi/pulumi-azure/sdk/v4/go/azure/eventgrid"
-	"github.com/pulumi/pulumi-azure/sdk/v4/go/azure/operationalinsights"
+
```
Comment (asalkeld) [inline pkg/provider/pulumi/azure/containerapp.go]: delete space ^

File: pkg/provider/pulumi/azure/containerapp.go [hunk 5]
```diff
@@ -98,43 +101,75 @@ func (a *azureProvider) newContainerApps(ctx *pulumi.Context, name string, args
...
-	aw, err := operationalinsights.NewAnalyticsWorkspace(ctx, resourceName(ctx, name, AnalyticsWorkspaceRT), &operationalinsights.AnalyticsWorkspaceArgs{
+	aw, err := operationalinsights.NewWorkspace(ctx, resourceName(ctx, name, AnalyticsWorkspaceRT), &operationalinsights.WorkspaceArgs{
 		Location:          args.Location,
 		ResourceGroupName: args.ResourceGroupName,
-		Sku:               pulumi.String("PerGB2018"),
-		RetentionInDays:   pulumi.Int(30),
+		Sku: &operationalinsights.WorkspaceSkuArgs{
+			Name: pulumi.String("PerGB2018"),
+		},
+		// Sku:               pulumi.String("PerGB2018"),
```
Comment (asalkeld) [inline pkg/provider/pulumi/azure/containerapp.go]: remove the commented out code

File: pkg/codeconfig/codeconfig.go (hunk 1 above)
Comment (tjholm) [reply pkg/codeconfig/codeconfig.go]: Currently doing a rebase, dropping changes to this file.

File: pkg/provider/pulumi/generator.go [hunk 6]
```diff
@@ -75,11 +75,17 @@ func New(s *project.Project, t *stack.Config) (types.Provider, error) {
...
 	*/
-	return &pulumiDeployment{
+	pDep := &pulumiDeployment{
 		s: s,
 		t: t,
 		p: prov,
-	}, nil
+	}
+
+	if err := pDep.p.Validate(); err != nil {
```
Comment (asalkeld) [inline pkg/provider/pulumi/generator.go]: this is been done now first thing in the cmd.Up

Comment (tjholm) [reply pkg/provider/pulumi/generator.go]: commit dropped from rebase

File: pkg/provider/pulumi/azure/containerapp.go (hunk 5 above)
Comment (tjholm) [reply pkg/provider/pulumi/azure/containerapp.go]: done

File: pkg/provider/pulumi/azure/containerapp.go (hunk 4 above)
Comment (tjholm) [reply pkg/provider/pulumi/azure/containerapp.go]: done

File: pkg/provider/pulumi/azure/azure.go (hunk 3 above)
Comment (tjholm) [reply pkg/provider/pulumi/azure/azure.go]: done

File: pkg/provider/pulumi/azure/azure.go (hunk 2 above)
Comment (tjholm) [reply pkg/provider/pulumi/azure/azure.go]: done

**Review by asalkeld — APPROVED** [review]
lgtm

**Comment by tjholm** [comment]
@asalkeld requesting a re-review as I've made some minor changes to the naming for storage containers and queues. I've tested this with two apps to confirm it's working.

Would also be good to bump the membrane version based on this as well: https://github.com/nitrictech/nitric/pull/250 as there are some changes in the membrane that are complimented by these fixes.

**Review by asalkeld — APPROVED** [review]

**Review by jyecusch — APPROVED** [review]

**Pull request merged**